wagtail/snippets/templates/wagtailsnippets/snippets/headers/_base_header.html

```diff
@@ -0,0 +1,42 @@
+{% extends 'wagtailadmin/shared/headers/slim_header.html' %}
+{% load wagtailadmin_tags i18n %}
+
+{% block header_content %}
+    {% with breadcrumb_link_classes='w-flex w-items-center w-h-full w-text-primary w-px-0.5 w-no-underline w-outline-offset-inside hover:w-underline hover:w-text-primary w-h-full' breadcrumb_item_classes='w-h-full w-flex w-items-center w-overflow-hidden w-transition w-duration-300 w-whitespace-nowrap w-flex-shrink-0 w-font-bold w-text-14' icon_classes='w-w-4 w-h-4 w-ml-3' %}
```

Comment:
@thibaudcolas  & @laymonage 

there is a separate discussion in this issue about whether breadcrumbs are suitable for snippets.
see https://github.com/wagtail/wagtail/issues/8645 (comments)

The tldr is - should breadcrumbs only really be for representing a Page tree in the admin UI and not the admin's page tree.

While this PR is still WIP it would be good to resolve the direction we want to head in.

Additional thoughts
* We should really try to avoid copy/pasting the entire breadcrumbs HTML here if we can avoid it - there is upcoming work starting next week to refactor the new breadcrumbs classes/styles and usage to be suitable in other parts of Wagtail that already use breadcrumbs
* If we do or do not do this - it would be good to add a note to https://github.com/wagtail/wagtail/issues/8597 with an update.

Reply:
> We should really try to avoid copy/pasting the entire breadcrumbs HTML here if we can avoid it - there is upcoming work starting next week to refactor the new breadcrumbs classes/styles and usage to be suitable in other parts of Wagtail that already use breadcrumbs

I agree. I initially wanted to make the breadcrumbs' template tag more generic (decoupling it from the Page tree logic), but we haven't really decided the extent of the new breadcrumbs.

> If we do or do not do this - it would be good to add a note to https://github.com/wagtail/wagtail/issues/8597 with an update.

Apart from that feature request, the main reason why I did this was because we need to bring the side panels for the new Snippets features (revisions, live/draft, etc.). Having just the side panel buttons on the header (without the breadcrumbs) doesn't look so good to me. But yeah, bringing this in means that the breadcrumbs also represent the admin's page tree, and it will be a bit inconsistent if we don't add this to other parts in the admin.